server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -352,6 +364,61 @@ private CompletionStage<RestResponse> report(RestRequest restRequest) {
       );
    }
 
+   private CompletionStage<RestResponse> nodeReport(RestRequest restRequest) {
+      EmbeddedCacheManager cacheManager = invocationHelper.getProtocolServer().getCacheManager();
+      List<Address> members = cacheManager.getMembers();
+      if (members == null) {
+         return report(restRequest);
+      }
+
+      String targetName = restRequest.variables().get("nodeName");
+      ByRef<ByteBuffer> response = new ByRef<>(null);
+      NettyRestResponse.Builder responseBuilder = new NettyRestResponse.Builder();
+      return SecurityActions.getClusterExecutor(cacheManager)
```

Comment:
we don't need to use the ```java invocationHelper.getExecutor()``` for the async calls ?

Reply:
I think it would be necessary in case of a blocking method.